### PR TITLE
jscheid/fix deep file recursion

### PR DESCRIPTION
**Pull request by joshuascheid**
- Fix instability resulting from +seed commandline use. (#9)
- Fix warnings with GCC 7.1 and later (which are errors due to -Werror). (#10)
- Fix stack overflow issue.

### CODE REVIEW

**Comment by joshuascheid** [comment]
@paul-donahue @jayantom

**Comment by paul-donahue** [comment]
Some of this stuff is in https://github.com/jayantom/sknobs/pull/9.  Can you merge upstream/master or click on "fetch upstream" next to "this branch is N commits behind ..." on github?  But I can look at individual commits rather than the overall PR changes.

File: src/c/sknobs.c [hunk 1]
```diff
@@ -921,21 +922,32 @@ int sknobs_load(int argc, char *argv[], char *comment) {
...
+  argv = malloc(ARGV_SIZE_CHUNK * sizeof(char*));
+  assert(argv);
+  current_argv_size = ARGV_SIZE_CHUNK;
   argv[argc++] = name;
   token = strtok(s, sknobs_delimiters);
   while (token) {
-    assert(argc < MAX_ARGV_SIZE);
+    if(argc >= current_argv_size) {
+      // Incrementally add chunks.
+      assert(argc < MAX_ARGV_SIZE);
```
Comment (paul-donahue) [inline src/c/sknobs.c:940]: This is OK given the current values of those #defines but abstractly (e.g. if MAX_ARGV_SIZE is not a multiple of ARGV_SIZE_CHUNK) it allows allocation of more than MAX_ARGV_SIZE and it also allows argc to iterate beyond MAX_ARGV_SIZE.  Not horrible but not perfect.

I was going to say "don't worry about it" but I got intrigued and think this works:
```c
    if(argc >= current_argv_size) {
      int newsize = current_argv_size + ARGV_SIZE_CHUNK;
      assert(current_argv_size < MAX_ARGV_SIZE);
      current_argv_size = (newsize <= MAX_ARGV_SIZE) ? newsize : MAX_ARGV_SIZE;
      argv = realloc(argv, current_argv_size * sizeof(char*));
      assert(argv);
    }
```
Or you can punt.  It's not really a practical problem.

Comment (joshuascheid) [reply src/c/sknobs.c:940]: What about defining MAX to be N chunks?

Comment (paul-donahue) [reply src/c/sknobs.c:940]: Yep.

**Comment by joshuascheid** [comment]
I'm going to move this to jayantom/sknobs

**Pull request closed without merging**